core/commands/swarm.go

```diff
@@ -135,6 +139,18 @@ var swarmPeeringAddCmd = &cmds.Command{
 				return err
 			}
 		}
+
+		save, _ := req.Options[swarmSaveOptionName].(bool)
+		if save {
+			update := func(cfg *config.Config) {
+				cfg.Peering.Peers = node.Peering.ListPeers()
```

Comment:
AFAIT this means that:
```
ipfs swarm peering add $M57
ipfs swarm peering add --save $NGC6543
```
Will save both `$M57` and `$NGC6543` in the config because they are both live when `--save` is triggered.

My instincts would be that only `$NGC6543` should be added.
Probably by something similar to `cfg.Peering.Peers = append(cfg.Peering.Peers, addInfos)`

Reply:
Good catch, fixed here [35efa59](https://github.com/ipfs/kubo/pull/9425/commits/35efa59ee656069a08f2aef466c922d3847410c0)